Add mugur recipe

mihaiolteanu opened the pull request:
### Brief summary of what the package does

A configurator for Ergodox and other qmk-powered keyboards.

### Direct link to the package repository

https://github.com/mihaiolteanu/mugur

### Your association with the package

Author and maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Neato, I have been coveting Ergodox keyboards for a while. :)

Quick first pass --

### ergodox-ez.el ###

This file is a little bit tricky.  Is it an example, or a feature that the user would load/unload (in which case, it resembles a minor mode to me, but I might not have all the context).  In the meantime --

byte-compile-file (using Emacs 26.3):
- Per CONTRIBUTING.org, consider using lexical binding.
  See: https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#fixing-typical-problems
- Lexical binding will be used in what follows
```
ergodox-ez.el:1:7:Warning: assignment to free variable `mugur-layout-vertical'
ergodox-ez.el:21:7:Warning: assignment to free variable
    `mugur-layout-horizontal'
```
I think you should have a `(require 'mugur)` in this file?

checkdoc (using version 0.6.1):
```
Warning (emacs): 
ergodox-ez.el:0: The first line should be of the form: ";;; package --- Summary"
Warning (emacs): 
ergodox-ez.el:0: You should have a section marked ";;; Commentary:"
Warning (emacs): 
ergodox-ez.el:1: You should have a section marked ";;; Code:"
Warning (emacs): 
ergodox-ez.el:43: The footer should be: (provide 'ergodox-ez)\n;;; ergodox-ez.el ends here
Warning (emacs): 
ergodox-ez.el:27: White space found at end of line
```
All of these assume that ergodox-ez.el is a feature and not just an example of course.

check-declare-file (optional):
- No issues!

package-lint-current-buffer (using version 20200419.406):
```
1 issue found:
1:0: error: There is no `provide' form.
```

### mugur.el ###

byte-compile-file (using Emacs 26.3):
```
mugur.el:151:11:Warning: reference to free variable `it'
In mugur--tapdance-extract:
mugur.el:324:21:Warning: reference to free variable `it'
In mugur--c-layer-state-set-user:
mugur.el:699:32:Warning: reference to free variable `it'
In mugur--generate-config-file:
mugur.el:805:31:Warning: reference to free variable `it'
In end of data:
mugur.el:913:1:Warning: the following functions are not known to be defined:
    mugur--keycode-string, hash-table-empty-p, awhen, mugur--set-keycodes,
    mugur--keycode-raw, mugur--key-in-category-p, mugur--special-key-p,
    mugur--modifier-key-p, mugur--keycode, mugur--key-or-sequence, aif,
    mugur--leds, mugur--keys, mugur--orientation, mugur--replace-custom-keys,
    mugur--keymaps-all
```
I think some flattening of the code might clean these up, but I haven't gotten my feet wet.

checkdoc (using version 0.6.1):
- No issues!

package-lint-current-buffer (using version 20200419.406):
```
1 issue found:
1:0: warning: The package summary should not end with a period.
```

### Loadability ###

Running #'load on each file:
```
Loading ergodox-ez.el
Loading mugur.el
Done.
```

mihaiolteanu commented:
**ergodox-ez.el**

Removed it. The idea was to support all the qmk keyboards and every such .el file would have the layout and all the necessary info for each keyboard. The qmk is rather inconsistent regarding the configs for different keyboards, so I've decided to only support the Ergodox since that is what I have and that is what I'm interested in. 

**mugur.el**

Fixed them all.. I hope.

> Neato, I have been coveting Ergodox keyboards for a while. :)

Not many resources and best practices for Emacs users out there for these keyboards, but lots of fans blinking LEDs and changing keycaps for fun.

I was rather led into it by wrist and back pain from all those awful, cheap plastic cases I've used over the years. It took a few months to figure out HOW to use it and then some months more to actually get use to typing on it.

riscy commented:
Thanks!
